src/modules/weather/weather.repository.ts

```diff
@@ -1,49 +1,17 @@
-import { firstValueFrom } from "rxjs";
-import { HttpException, Injectable } from "@nestjs/common";
-import { HttpService } from "@nestjs/axios";
-import { AxiosError } from "axios";
-import { ErrorMessage, ErrorStatusCode } from "../../libs/enums/enums.js";
+import { WeatherDto } from "./types/types.js";
+
 import {
-  WeatherApiResponseDto,
-  WeatherConfig,
-  WeatherDto,
-  WeatherError,
-} from "./types/types.js";
-import { WeatherErrorCode } from "./enums/enums.js";
-import { IWeatherRepository } from "./interfaces/interfaces.js";
+  IWeatherProvider,
+  IWeatherRepository,
+} from "./interfaces/interfaces.js";
+import { Injectable } from "@nestjs/common";
 
 @Injectable()
 class WeatherRepository implements IWeatherRepository {
-  public constructor(
-    private readonly httpService: HttpService,
-    private readonly config: WeatherConfig
-  ) {}
-
-  public async get(city: string): Promise<WeatherDto | null> {
-    try {
-      const { data } = await firstValueFrom(
-        this.httpService.get<WeatherApiResponseDto>(this.config.apiUrl, {
-          params: { key: this.config.apiKey, q: city },
-        })
-      );
-
-      return {
-        description: data.current?.condition.text,
-        humidity: data.current?.humidity,
-        temperature: data.current?.temp_c,
-      };
-    } catch (error: unknown) {
-      const axiosError = error as AxiosError<WeatherError>;
-
-      if (axiosError.response?.data.error.code === WeatherErrorCode.NOT_FOUND) {
-        return null;
-      }
+  constructor(private readonly firstProvider: IWeatherProvider) {}
```

Comment:
краще не надавати лишній контекст для класу
```suggestion
  constructor(private readonly provider: IWeatherProvider) {}
```
Але можливо й варто зробити WeatherProvider-и(BaseWeatherProvider) implements IWeatherRepository, адже тепер цей клас є простой проксьою і не несе в собі іншого призначення